src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -242,5 +241,32 @@ internal SocketConnection(Socket socket, MemoryPool<byte> memoryPool, ISocketsTr
 
             return error;
         }
+
+        // This scheduler runs on the global thread pool queue instead of the local queue
+        // We can delete this when we get a new pipelines build since that's the new default.
+        // In the current version that Kestrel uses, we're using local queues instead of the global one.
+        private class GlobalThreadPoolScheduler : PipeScheduler
+        {
+            public static GlobalThreadPoolScheduler Default = new GlobalThreadPoolScheduler();
+
+            public override void Schedule(Action action)
+            {
+                System.Threading.ThreadPool.QueueUserWorkItem(s =>
+                {
+                    ((Action)s)();
+                },
+                action);
+            }
+
+            public override void Schedule(Action<object> action, object state)
+            {
+                System.Threading.ThreadPool.QueueUserWorkItem(s =>
+                {
+                    var t = (Tuple<Action<object>, object>)s;
+                    t.Item1(t.Item2);
+                },
+                Tuple.Create(action, state));
```

Comment:
Tuple?

Reply:
For now this is only used per connection so not a big deal